aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -468,8 +474,13 @@ module aptos_framework::fungible_asset {
     /// Check whether the balance of a given store is >= `amount`.
     public fun is_balance_at_least<T: key>(store: Object<T>, amount: u64): bool acquires FungibleStore {
         let store_addr = object::object_address(&store);
-        if (store_exists(store_addr)) {
-            borrow_store_resource(&store).balance >= amount
+        is_address_balance_at_least(store_addr, amount)
+    }
+
+    /// Check whether the balance of a given store is >= `amount`.
+    public(friend) fun is_address_balance_at_least(store_addr: address, amount: u64): bool acquires FungibleStore {
```

Comment:
I don't see why this has to be friend only. Let's just keep it public?

Reply:
we have public is_balance_at_least<T>(store: Object<T>)

it's not clear for usecases other than gas charging, that it is worth optimization this brings, or that we want to encourage people to not use objects.
keeping things public(friend) means we can also remove them later. public thing is forever